packages/core/src/types/IViewport.ts

```diff
@@ -6,19 +6,99 @@ import { ActorEntry } from './IActor';
 import ViewportType from '../enums/ViewportType';
 import ViewportStatus from '../enums/ViewportStatus';
 import DisplayArea from './displayArea';
+import BoundsLPS from './BoundsLPS';
 
 /**
  * Specifies what image to get a reference for.
  */
 export type TargetSpecifier = {
   /** The slice index within the current viewport camera to get a reference for */
-  sliceIndex?: number;
-  /** True to get a frame of reference UID reference instead of a regular image one */
+  sliceIndex?: number | [number, number];
+  /**
+   * true to get a frame of reference UID reference instead of a regular image one,
+   * or false to not get it when it would otherwise do so.
+   * The default depends on the view type, but in general showing multiple images
+   * combined, the return will be only for frame of reference, while showing a
+   * single one, it will be for frame of reference.
+   */
   forFrameOfReference?: boolean;
   /** Set of points to get a reference for, in world space */
   points?: Point3[];
   /** The volumeId to reference */
   volumeId?: string;
+  /**
+   * true to include the bounds, if it would not otherwise do so, or
+   * false to exclude them if it would normally do so.
+   * Bounds are normally only included if the viewport is very magnified
+   * AND the points are included in the request, or the view is for a range
+   * of slices.
+   */
+  includeBounds?: boolean;
+};
+
+export type ViewCompatibleOptions = {
+  /** Indicate if the test should proceed if the view were navigated */
+  withNavigation?: boolean;
+  /**
+   * Test to see if this view could be shown in the volume equivalent (eg by
+   * changing the normal etc)
+   */
+  asVolume?: boolean;
+  /**
+   * Use this imageURI for testing - may or may not be the current one.
+   * Should be a straight contains URI for the set of imageIds in any of
+   * the volumes or set of image ids.
+   */
+  imageURI?: string;
+};
```

Comment:
This is not really clear to me the purpose of these options.

Reply:
Updated the docs quite a bit.